packages/flutter/lib/src/gestures/binding.dart

```diff
@@ -225,6 +261,47 @@ mixin GestureBinding on BindingBase implements HitTestable, HitTestDispatcher, H
       pointerSignalResolver.resolve(event);
     }
   }
+
+  void _scheduleFrameCallback() {
+    if (_frameCallbackScheduled) {
+      return;
+    }
+    final SchedulerBinding? scheduler = SchedulerBinding.instance;
+    assert(scheduler != null);
+    if (scheduler != null) {
+      _frameCallbackScheduled = true;
+      scheduler.scheduleFrameCallback((_) {
+        _frameCallbackScheduled = false;
+        _sampleTime = scheduler.currentSystemFrameTimeStamp + samplingOffset;
```

Comment:
According to https://api.flutter.dev/flutter/scheduler/SchedulerBinding/currentSystemFrameTimeStamp.html, `currentSystemFrameTimeStamp`: 
> On most platforms, this is a more or less arbitrary value, and should generally be ignored. On Fuchsia, this corresponds to the system-provided presentation time, and can be used to ensure that animations running in different processes are synchronized.
which sounds like we should use `currentFrameTimeStamp` instead. (so it seems we don't need the `samplingOffset`?

Reply:
It's critical that this time stamp is in the same clock as the time stamps from input events. And it's also critical that this is never adjusted or scaled like currentFrameTimeStamp. This is something I [brought up](https://github.com/flutter/engine/pull/15936#issuecomment-579026468) when we were discussing if this code should live in the framework or in the engine. I was told that we can rely on currentSystemFrameTimeStamp working like Fuchsia on other platforms. Maybe this documentation needs to be updated.

Reply:
Can you add that information as a comment here?

Reply:
Done